src/subscript/fmu_copy_revision/fmu_copy_revision.py

```diff
@@ -0,0 +1,654 @@
+- rms/output
+- rms/input/seismic/
+- share/results/
+- share/templates/
++ */
+- *
+"""
+
+DIRFILTER5 = """
+- backup/
+- users/
+- attic/
+- .git/
+- ert/output/
+- ert/storage/
+- ert/**/storage/
+- rms/input/seismic/
++ */
+- *
+"""
+
+DIRFILTERX = """
++ */
+- *
+"""
+
+FILTER5 = FILTER2 + FILTER3_ADD + FILTER5_ADD
+FILTER3 = FILTER2 + FILTER3_ADD
+
+FILTER6 = """
++ share/coviz/**
+- *
+"""
+
+
+SHELLSCRIPT = """\
+#!/usr/bin/sh
+
+# SETUP OPTIONS
+
+SRCDIR="$1"  # a relative path
+DESTDIR="$2"  # must be an absolute path!
+FILTERFILE="$3"
+THREADS=$4
+RSYNCARGS="$5"
+DIRTREE=$6  # if 1 first copy folder tree, if 2 do it afterwards with dirfilterfile
+DIRFILTERFILE="$7"
+
+PWD=$(pwd)
+
+start=`date +%s.%N`
+
+cd $SRCDIR
+
+echo " ** Target folder is $DESTDIR"
+echo " ** Filter file is $FILTERFILE"
+mkdir -p $DESTDIR
+
+echo " ** Sync folders and files!"
+
+if [ $DIRTREE -eq 1 ]; then
+    echo " ** Sync all folders first..."  # this is usually fast
+    rsync -a -f"+ */" -f"- *" . $DESTDIR
+fi
+
+echo " ** Sync files using multiple threads..."
+find -L . -type f | xargs -n1 -P$THREADS -I% \
+    rsync $RSYNCARGS -f"merge $FILTERFILE" % $DESTDIR
+
+if [ $DIRTREE -eq 2 ]; then
+    echo " ** Sync all folders (also empty) except some..."
+    rsync -a -f"merge $DIRFILTERFILE" . $DESTDIR
+fi
+
+end=`date +%s.%N`
+
+runtime=$( echo "$end - $start" | bc -l )
+
+echo " ** Compute runtime..."
+
+echo $runtime
+cd $PWD
+
+"""
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """Setup parser."""
+
+    usetext = "fmu_copy_revision <commandline> OR interactive"
+    parser = argparse.ArgumentParser(
+        description=DESCRIPTION,
+        formatter_class=argparse.RawTextHelpFormatter,
+        usage=usetext,
+    )
+
+    parser.add_argument("--dryrun", action="store_true", help="Run dry run for testing")
+    parser.add_argument(
+        "--all, -all, -a", action="store_true", dest="all", help="List all folders"
+    )
+    parser.add_argument(
+        "--verbosity, -v",
+        action="store_true",
+        dest="verbosity",
+        help="Enable logging (messages) for debugging",
+    )
+    parser.add_argument(
+        "--cleanup",
+        action="store_true",
+        dest="cleanup",
+        help="Remove (cleanup) any existing target if present, default is False",
+    )
+    parser.add_argument(
+        "--skipestimate, --skip",
+        "-s",
+        action="store_true",
+        dest="skipestimate",
+        help="If present, skip estimation of current revision size.",
+    )
+    parser.add_argument("--source", dest="source", type=str, help="Add source folder")
+    parser.add_argument("--target", dest="target", type=str, help="Add target folder")
+    parser.add_argument(
+        "--profile",
+        dest="profile",
+        type=str,
+        default="3",
+        help="profile for copy profile to use, default is 3",
+    )
+    parser.add_argument(
+        "--threads",
+        dest="threads",
+        type=str,
```

Comment:
integer? If a  string is passed, it says "Do copying using foobar CPU threads, wait..." (everything seems to work, but an error should probably be issued).